vscode/src/completions/context/utils.ts

```diff
@@ -17,3 +17,39 @@ export function baseLanguageId(languageId: string): string {
             return languageId
     }
 }
+
+// On Node.js, an abort controller is implemented with the default event emitter
+// helpers which will warn when you add more than 10 event listeners. Since the
+// graph context implementation can fan out to many more concurrent requests, we
+// easily reach that limit causing a lot of noise in the console. This is a
+// lightweight abort controller implementation that does not have that limit.
+export class CustomAbortController {
+    private signal = new CustomAbortSignal()
+
+    public abort(): void {
+        this.signal.abort()
+    }
+}
+export class CustomAbortSignal {
+    private listeners: Set<() => void> = new Set()
+    public aborted = false
+
+    public addEventListener(eventName: 'abort', listener: () => void): void {
+        if (this.aborted) {
+            listener()
+            return
+        }
+    }
+
+    public removeEventListener(listener: () => void): void {
+        this.listeners.delete(listener)
+    }
+
+    public abort(): void {
+        this.aborted = true
```

Comment:
```suggestion
        if (this.aborted) return;  // Ensure abort is idempotent
        this.aborted = true
```

Ideally if abort is called multiple times, the listeners are should be notified the first time.